### DISCUSSION TITLE
Want to add background to transparent png

### DISCUSSION THREAD

**Discussion by jkgao** [ImageMagick/ImageMagick#2605] [Help]
I have a few hundred transparent png icons that I want to add a background to. The background is solid color and I have it in a png form or I can use the hex code if that works better. But what I want to do is batch add the same colored background to all these icons and have them saved as png. I tried using some of the resources online but I couldn't get it to work. I'm on Windows with the latest version of ImageMagick downloaded from the website. 

![One of the snippets of code I tried. Don't know if I'm in the right direction or not](https://user-images.githubusercontent.com/5335323/93828673-0ec6f500-fc29-11ea-9f2b-04d8667ecaa7.png)

**Comment by fmw42** [comment]
Your syntax is wrong. See https://imagemagick.org/Usage/basics/#mogrify_compose. The background image goes inside the -draw command in single quotes and the draw command has double quotes. Also inside the draw is the image flag: -draw "image 0,0, 'background_file.png'"

But an easier way is to just replace transparency with some color

`magick mogrify -format png -background red -alpha background -alpha off *.png`

If you want them to go to some empty directory, then add `-path path_to_your_output_directory`
